modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -298,10 +298,7 @@ def length(self):
             The length of the object.
         """
         if self._length_cache is None:
-            cls = type(self)
-            func = cls._length_extraction_fn()
-            preprocessed_func = cls.preprocess_func(func)
-            self._length_cache = self.apply(preprocessed_func)
+            self._length_cache = self.apply(self._length_extraction_fn()).get()
```

Comment:
If some execution relies on this implementation of `length`/`width` and at the same time has `preprocess_func`, which is useful for perf, then we may have perf penalties here.

Reply:
If I understand correctly, it helps if we apply this function several times in a row, but this is not the case. Also, if it's really needed, it's better to do it in the `apply` function itself.

Reply:
> If I understand correctly, it helps if we apply this function several times in a row, but this is not the case. Also, if it's really needed, it's better to do it in the `apply` function itself.

for the current engines, yes

> Also, if it's really needed, it's better to do it in the apply function itself.

not sure if it is better in the apply

anyway, let's leave as is for now